pkg/flavor/flavor.go

```diff
@@ -80,6 +92,90 @@ func (m *methods) ApplyToVmi(field *k8sfield.Path, profile *flavorv1alpha1.Virtu
 	return conflicts
 }
 
+func (m *methods) CheckFlavorRevision(vm *virtv1.VirtualMachine) error {
+	oldVM, err := m.client.VirtualMachine(vm.Namespace).Get(vm.Name, &v1.GetOptions{})
+	if k8serror.IsNotFound(err) {
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
+
+	if oldVM.Spec.FlavorRevision != nil && vm.Spec.FlavorRevision != nil {
+		if oldVM.Spec.FlavorRevision.Name != vm.Spec.FlavorRevision.Name {
+			return errors.New("flavor revision name is not the same as in older VM version")
+		}
+		if oldVM.Spec.FlavorRevision.Namespace != vm.Spec.FlavorRevision.Namespace {
+			return errors.New("flavor revision namespace is not the same as in older VM version")
+		}
+	}
+
+	return nil
+}
+
+func (m *methods) CreateFlavorRevision(vm *virtv1.VirtualMachine) error {
+	// don't modify flavor revision if it already exists
+	if vm.Spec.FlavorRevision != nil {
+		return nil
+	}
+
+	if vm.UID == "" {
+		return nil
+	}
+
+	flavor, err := m.FindProfile(vm)
```

Comment:
There's nothing guaranteeing that the profile found here is the exact same one that has been validated against.

The validation webhook needs to call `m.FindProfile` exactly once, and use that result to both validate the flavor applies to the VM and then set the flavor revision using the exact flavor that was validated.

When a flavor revision already exists, m.FindProfile should find the revision and use that for validation instead of anything else.

Reply:
Issue is, that I can't change vm definition (e.g. add flavor revision name) in validation webhook. I can create new controller revision from there, but I set the name of revision to vm and then update the vm. Is there any other way, how to update VM?

Reply:
set the flavor revision in the mutation webhook (create revision + set on vm.Status) and then validate using the revision in the validation webhook.